trl/models/utils.py

```diff
@@ -170,7 +172,13 @@ def add_hooks(model: "DeepSpeedEngine") -> None:
         optimizer_offload = model.optimizer.parameter_offload
     elif model.optimizer is not None:
         optimizer_offload = model.optimizer
-    optimizer_offload._register_hooks_recursively(optimizer_offload.module)
+    else:
+        raise NotImplementedError(f"Didn't yet handle optimizer {model.optimizer}.")
+    if deepspeed.__version__.split(".") >= ("0", "16", "4"):
```

Comment:
```suggestion
    if version.parse(deepspeed) >= version.parse("0.16.4"):
```

Reply:
you'll need to import version as well

Reply:
Just implemented this, thank you for the suggestion!